### DISCUSSION TITLE
[mobile] is possible to create a library for Android, iOS and web (JS) ?

### DISCUSSION THREAD

**Discussion by lukasMega** [tauri-apps/tauri#11702] [Q&A]
Is possible to have one code base for simple library and then compile it into following targets:
- iOS
- Android
- JS/TS for web app
?

More description:
I want to create a small library (or SDK) that offers basic functionality for HTTP requests and WebSocket communication, utilizing an HTTP REST API for creating and processing requests and responses. The goal is to provide a common API for the various platforms mentioned above. This library should be compiled into platform-native code and designed for ease of use, ensuring it is as straightforward to work with as other libraries.

I'm not sure whether Tauri is a good fit for it.

**Comment by FabianLars** [comment]
> I'm not sure whether Tauri is a good fit for it.

Tauri is more for building (end user) apps, not libraries. It also can't target the web itself.
